nucleus/common/glassfish-api/src/main/java/org/glassfish/api/jdbc/SQLTraceRecord.java

```diff
@@ -194,21 +218,107 @@ public void setParams(Object[] params) {
+     *
+     * @return Module name
+     */
+    public String getModuleName() {
+        return moduleName;
+    }
+
+    /**
+     * Set the module name which executed the SQL statement
+     *
+     * @param moduleName
+     */
+    public void setModuleName(String moduleName) {
+        this.moduleName = moduleName;
+    }
+
+    /**
+     * Get the stack frame for the method call in the application that triggered SQL execution, if the call comes from a deployed application.
+     *
+     * This returns a frame which returns a class from the {@link StackWalker.StackFrame#getDeclaringClass()} method
+     * ({@link StackWalker.Option.RETAIN_CLASS_REFERENCE} is enabled).
+     *
+     * @return Stack frame that represents a call to a server component that triggered SQL execution
+     */
+    public Optional<StackWalker.StackFrame> getCallingApplicationMethod() {
+        return Optional.ofNullable(callingApplicationMethod);
+    }
+
+    /**
+     * Set the stack frame for the method call in the application that triggered SQL execution.
+     * The {@link StackWalker.StackFrame#getDeclaringClass()} should return a class and never throw {@link UnsupportedOperationException}.
+     */
+    public void setCallingApplicationMethod(StackWalker.StackFrame callingApplicationMethod) {
+        this.callingApplicationMethod = callingApplicationMethod;
+    }
+
+
+
     @Override
     public String toString() {
-        StringBuffer sb = new StringBuffer();
-        sb.append("ThreadID=" + getThreadID() + " | ");
-        sb.append("ThreadName=" + getThreadName() + " | ");
-        sb.append("TimeStamp=" + getTimeStamp() + " | ");
-        sb.append("ClassName=" + getClassName() + " | ");
-        sb.append("MethodName=" + getMethodName() + " | ");
+        StringBuilder sb = new StringBuilder();
+        sb.append("PoolName=").append(getPoolName()).append(" | ");
+        sb.append("ThreadID=").append(getThreadID()).append(" | ");
+        sb.append("ThreadName=").append(getThreadName()).append(" | ");
+        sb.append("TimeStamp=").append(getTimeStamp()).append(" | ");
+        sb.append("SQL=").append(getSqlQuery()).append(" | ");
+        sb.append("AppName=").append(getApplicationName()).append(" | ");
+        sb.append("ModuleName=").append(getModuleName()).append(" | ");
+        sb.append("ClassName=").append(getClassName()).append(" | ");
+        sb.append("MethodName=").append(getMethodName()).append(" | ");
         if (params != null && params.length > 0) {
             int index = 0;
             for (Object param : params) {
-                sb.append("arg[" + index++ + "]=" + (param != null ? param.toString() : "null") + " | ");
+                sb.append("arg[").append(index++).append("]=")
+                        .append(param != null ? param.toString() : "null").append(" | ");
             }
         }
-        // TODO add poolNames and other fields of this record.
+        if (callingApplicationMethod != null) {
```

Comment:
Just invert the if, so the condition would be without negation. It is easier to read.